### PR TITLE
Allow 'Background' to be used in 'Rule'

### PR DESCRIPTION
**Pull request by BerhorShadowbuckle**
According to https://cucumber.io/docs/gherkin/reference/ `Background` can be used in `Rule` (it's not recommended, but it's allowed), so this commit enables such usage.

### CODE REVIEW

**Comment by bbqsrc** [comment]
Hmm, I recall reading from a patch on the cucumber repo that they removed support for this in the mainline Cucumber implementation.

Yes, the reference is regularly incorrect and it drives me mad. I am leaning toward merging this, though first I will look up whether it is actually supported in mainline Cucumber and document that here, so I can look back and know why i did this. 😄 

Thanks for another clean contribution!

**Comment by bbqsrc** [comment]
Found the issues:

- https://github.com/cucumber/cucumber/issues/590
- https://github.com/cucumber/docs.cucumber.io/pull/330
- https://github.com/cucumber/docs.cucumber.io/issues/307

tl;dr it has been banned and unbanned multiple times, haha.

As far as I am concerned, whether the Gherkin grammar supports it or not is separate to whether the Cucumber tool will support it, so I will merge this.

**Comment by bbqsrc** [comment]
Published in 0.8.3. Thanks again!
